src/averageAges.js

```diff
@@ -55,7 +71,17 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const child = people.filter(person => onlyWithSon
+    ? person.sex === 'm' && people.some(male => male.name === person.mother)
+    : people.some(male => male.name === person.mother));
+
+  const ages = child.map(person => {
+    const mom = people.filter(male => male.name === person.mother);
```

Comment:
You can use the 'find' method instead of 'filter' for finding the mother, as it returns the first element that matches the condition.